src/lib/converter/utils/types.ts

```diff
@@ -0,0 +1,45 @@
+import * as ts from 'typescript';
+
+/**
+ * Returns the type parameters of a given type.
+ * @param type The type whos type parameters are wanted.
+ * @returns The type parameters of the type. An empty array if the type has no type parameters.
+ */
+export function getTypeParametersOfType(type: ts.Type): ts.TypeParameterDeclaration[] {
+    for (const declaration of type.symbol.declarations) {
```

Comment:
`symbol` might be undefined, and `symbol.declarations` might be undefined, leading to a crash, prefer `type.getSymbol()?.getDeclarations() ?? []`